src/pdm/utils.py

```diff
@@ -69,10 +72,7 @@ def url_without_fragments(url: str) -> str:
 
 
 def join_list_with(items: list[Any], sep: Any) -> list[Any]:
-    new_items = []
-    for item in items:
-        new_items.extend([item, sep])
-    return new_items[:-1]
+    return list(iter_chain.from_iterable(iter_product(items, [sep])))[:-1]
```

Comment:
This brings more burden to understand, I prefer to keep the old code.

Reply:
OK, no problem. Will revert

Reply:
These test failures relate to path processing, which obviously works differently on Windows and POSIX. I'll need to revisit some of the test data here.

Reply:
@frostming I think I've fixed the test failures. There were issues here: (1) incompatibility in test path instantiation between POSIX and Windows platforms which affected `utils.get_venv_like_prefix` and `utils.path_to_url`, (2) incompatibility in path equality checking between pre-Python 3.12.x and Python 3.12.x versions which affected `utils.get_venv_like_prefix`.

I have Python 3.12.a01 installed via `pyenv`, but was unable to install PDM(2.9.3) due to an error, so could not test the changes locally in 3.12. But they should now work.

Reply:
path comparison is still failing. Thanks